### ISSUE TITLE
0.11.0 build fails on macOS due to unimplemented map_extents

### ISSUE THREAD

**Issue by dsully** [tarka/xcp#38]
While building on macOS with `rustc 1.71.1 (eb26296b5 2023-08-03)`

```
   Compiling xcp v0.11.0
error[E0308]: mismatched types
   --> /Users/dsully/.cargo/registry/src/index.crates.io-6f17d22bba15001f/xcp-0.11.0/src/drivers/parblock.rs:145:16
    |
145 |         if let Some(extents) = map_extents(&harc.infd)? {
    |                ^^^^^^^^^^^^^   ------------------------ this expression has type `Vec<std::ops::Range<u64>>`
    |                |
    |                expected `Vec<Range<u64>>`, found `Option<_>`
    |
    = note: expected struct `Vec<std::ops::Range<u64>>`
                 found enum `Option<_>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `xcp` (bin "xcp") due to previous error
error: failed to compile `xcp v0.11.0`, intermediate artifacts can be found at `/var/folders/q3/mk5rnvmd7rg59mgr20by9b0h0000gn/T/cargo-installZgCPLo`
```

This is due to `map_extents` in `src/os/common.rs`:

```rust
#[allow(dead_code)]
pub fn map_extents(_fd: &File) -> Result<Vec<Range<u64>>> {
    // FIXME: Implement for *BSD with lseek?
    Err(XcpError::UnsupportedOperation {}.into())
}
```

And is due to the change in e4d8cd2, and related to issue #37

**Comment by tarka** [comment]
Fixed in 0.12.0, and Mac testing has been cleaned-up and fixed.

**Comment by tarka** [comment]
NOTE: I no longer have access to a Mac, so support should be considered best-effort.
